downscale/estimate_forcing_regression.f90

```diff
@@ -461,14 +526,31 @@ end subroutine heapsort
     ! do power transformation on precip vector (AWW: consider alternate transforms)
     call normalize_y (4.0d0, y)    ! SHOULD NOT BE HARDWIRED
 
+
+    !read from nwp predictor file list
+    read(unit=55,fmt='(A)',iostat=error) nwp_timestep_file
+    print *,trim(nwp_timestep_file)
+    if(error /= 0) then
+      print *, 'Error reading from NWP predictor file list: ',trim(nwp_input_list)
+      stop
+    end if
+
+    !read NWP predictor file for current timestep
+    call read_nwp(times(t),nwp_timestep_file,nPredict,n_nwp,nwp_vars,nearestGridpoint,x,z,error)
+    if(error /= 0) then
+      print *, 'Error in read_nwp'
+      stop
+    end if
+
+
     ! -------- loop through all grid cells for a given time step --------
     do g = 1, ngrid, 1
       ! call system_clock(tg1,count_rate)
 
       deallocate (twx_red)
       deallocate (tx_red)
-      allocate (twx_red(6, sta_limit))! these have dim1 = 6
-      allocate (tx_red(6, sta_limit))
+      allocate (twx_red(nPredict, sta_limit))! these have dim1 = 6
```

Comment:
these are some of the allocations & deallocations that I suspect are redundant.  I went through a big cleanout much earlier, but it seems that if memory space is allocated once for these variables, we shouldn't have to deallocate and reallocate it inside of loops.  there may be more do clear out.